torch/csrc/autograd/variable.cpp

```diff
@@ -21,18 +21,20 @@
 
 namespace torch {
 namespace autograd {
-Variable::Impl::Impl(at::Tensor data, bool requires_grad, Edge gradient_edge)
+Variable::Impl::Impl(at::Tensor data, bool requires_grad, Edge gradient_edge, Variable::AutogradMeta* autograd_meta)
     : TensorImpl(data.type_id(), data.dtype(), /*allocator=*/nullptr, /* is variable */ true),
-      data_(std::move(data)),
-      grad_fn_(std::move(gradient_edge.function)),
-      requires_grad_(false),
-      is_view_(false),
-      output_nr_(gradient_edge.input_nr),
-      pyobj_(nullptr) {
+      data_(std::move(data)) {
+  autograd_meta->grad_fn_ = std::move(gradient_edge.function);
```

Comment:
Hmm... if you're going to set the fields of `autograd_meta` in this constructor, why don't you just construct it in the constructor to begin with?